### DISCUSSION TITLE
for placing text on interval rect bars, is there an easy option to align text center to center of the bar? e.g. offset to half width of the interval rect bar?

### DISCUSSION THREAD

**Discussion by tx0c** [observablehq/plot#611] [Q&A] [answered]
I'm doing a month graph with rect with { x, interval }, want to place text on top of each bar at center, if I don't specify `textAnchor` then text is placed center aligning the left side of the bar, not desired; then I use `textAnchor: "start",` but can only align start of text to left of the bar;
this monthly rect bar is similar to `Plot.barX` chart but difference is each month bar is a little bit different width, e.g. most February has 28 days vs other months' 30 or 31 days width, so February is always a bit narrower, hard to align it with just `dx` hard-coded offset;

```js
    Plot.rectY(tagsAllPerMonth, {
      x: "month", interval: d3.utcMonth, y: "count",
      stroke: "black", rx: 8, }),
    Plot.text(tagsAllPerMonth, {
      x: "month", interval: d3.utcMonth, y: "count",
      text: "count", dy: -6, textAnchor: "start",
    }),
```

![image](https://user-images.githubusercontent.com/49659410/145698408-b5fd8ec3-6ffe-4248-9b1d-6f0cad2edf5b.png)

I think the correct `dx` should be a function like `      dx:    d => (x(NextMonth) - x(CurrentMonth)) / 2, ` but still have not figured out a correct expression, where can I get a reference of the domain mapping function `x` ?

```js
    Plot.text(tagsAllPerMonth, {
      x: "month", interval: d3.utcMonth, y: "count",
      text: "count", dy: -6, // textAnchor: "start", // leave textAnchor default to "center"
      dx:    d => (xNextMonth - xCurrentMonth) / 2,
    }),
```

**Comment by Fil** [comment, marked as answer]
Plot.textY doesn't have the interval transform (https://github.com/observablehq/plot/issues/597), but you might want to do one of:

_add 14 days:_
~~~js
x: d => +d3.utcMonth.floor(d["date_of_birth"]) + 14 * 24 * 3600 * 1000
~~~

_compute the middle of the month:_ 
~~~js
x: d => (+d3.utcMonth.floor(d["date_of_birth"]) + +d3.utcMonth.ceil(1 + +d["date_of_birth"])) / 2
~~~

see https://observablehq.com/@fil/plot-center-text-on-an-interval
